server/src/main/java/org/elasticsearch/action/admin/cluster/storedscripts/GetScriptContextResponse.java

```diff
@@ -28,69 +28,55 @@
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.script.ScriptContextInfo;
 
 import java.io.IOException;
-import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.stream.Collectors;
 
-import static org.elasticsearch.common.xcontent.XContentParser.Token.END_OBJECT;
-import static org.elasticsearch.common.xcontent.XContentParser.Token.START_OBJECT;
-
 public class GetScriptContextResponse extends ActionResponse implements StatusToXContentObject {
 
     private static final ParseField CONTEXTS = new ParseField("contexts");
-    private final List<String> contextNames;
+    final Set<ScriptContextInfo> contexts;
```

Comment:
Why is this a Set? Seems like ti should be a map, from the context name, to the context info.

Reply:
My thinking was that the `Map` was just a `Set` with redundant info.  Changed to `Map`.